### PR TITLE
Revise incompatibility warning FAQs

### PR DESCRIPTION
**Pull request by fynngodau**
I implemented the ideas from #1288 and split up the `incompatibility_warning` article into `part_incompat` and `full_incompat`.

To avoid duplicate fixes in response to your feedback, I am only providing an English version until all your concerns with this version are addressed.

image

image

I added a redirect from `incompatibility_warning` to `part_incompat`.

### CODE REVIEW

File: src/data/faq.json
Comment (Ein-Tim) [inline src/data/faq.json:501]: Maybe add a link to https://support.google.com/android/answer/3094742?hl=en&ref_topic=7313011 to "contact your device manufacturer"?

File: src/data/faq.json
Comment (Ein-Tim) [inline src/data/faq.json:502]: I'd say "are fully functional", since we're talking about more than one feature, or?

File: src/data/faq.json
Comment (Ein-Tim) [inline src/data/faq.json:516]: Reg. "contact your device manufacturer", see my other comment.

File: src/data/faq.json
Comment (Ein-Tim) [inline src/data/faq.json:517]: Same here reg. "is" -> "are"

**Review by Ein-Tim — COMMENTED** [review]
One grammatically point and one suggestion. Entry looks good to me! 
Nice that you added the redirect!

File: src/data/faq.json
Comment (fynngodau) [reply src/data/faq.json:501]: It seems like a helpful resource to me, but I'm not sure whether we can link it and would like to have an official confirmation before implementing this.

File: src/data/faq.json
Comment (fynngodau) [reply src/data/faq.json:502]: From my feeling, since I'm separating the first batch and "retrieving […] test results" through an "as well as", the verb would grammatically refer only to the test result feature, but honestly I have no idea how to find out which version is correct :sweat_smile:

Comment (Ein-Tim) [reply src/data/faq.json:502]: Maybe @MikeMcC399 would like to help out here? 🙂

File: src/data/faq.json
Comment (Ein-Tim) [reply src/data/faq.json:501]: Okay, let's wait for confirmation then. This is already linked to from https://www.coronawarn.app/en/faq/#various_device_problems, so shouldn't be a problem, I assume.

**Comment by heinezen** [comment]
I have a few general remarks for improvement
----
The error message ("incompatibility warning") should be part of the entry heading. This way, search engines will hopefully lead  users directly to the FAQ entry if they search for the warning online.

----

Using a more active language and shorter sentences is encouraged

Example:

> Since version 2.2, the Corona-Warn-App for Android contains a check which determines whether your device is fully capable of all Bluetooth-related features which the app needs to operate normally.

can be improved to

> Since version 2.2, the Corona-Warn-App **checks** if your device **supports** all Bluetooth-related features normally used by the app.

----
If possible, we should get straight to the point

Example:

> If you are seeing this message, the operating system of your device is lacking a functionality that Corona-Warn-App normally uses.

we can merge that sentence with the next paragraph

> If you are seeing this message, the operating system of your device **lacks the ability to send random IDs** [to devices in your proximity]. [...]

The reason is that we want to get to the ">your own risk assesment is not affected" part quickly.

----
"Schachtelsatz" construction should be avoided, especially in english, but it's also a good idea for the german entry.

> The only limitation which this incompatibility causes is that, in case you are infected and you decide to warn others, only users who were checked in to the same event or the same location, at the same time as you were, can be warned.

The way we could split this is by separating the information communicated here into separate sentences:

1. What is the limitation/What is limited? (warning people who were near you)
2. What does still work? (warning through Check-In functionality)

File: src/data/faq.json
Comment (heinezen) [reply src/data/faq.json:501]: It's a good idea to include the link IMO

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:500]: Suggest:
"All other features of the app are not affected." =>
"None of the other app features are affected."

File: src/data/faq.json
Comment (MikeMcC399) [reply src/data/faq.json:502]: This sentence seems redundant, because the message has already been given in L500. 

Grammatically it is a little awkward because the main sentence says:
"Both Check-in and Contact Journal functionality is fully functional without restriction and only proximity-based ID recording is affected."
functionality / functional appears twice and "fully" and "without restriction" is a repeat.

I would tend to just drop this sentence.

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:514]: Add after L514
"The risk assessment cannot take account of other close-by smartphones because it cannot detect them."

File: src/data/faq.json
Comment (MikeMcC399) [reply src/data/faq.json:517]: Suggest also here leaving out this problematic sentence. The message should be covered by L514.

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
I made some suggestions for change.

File: src/data/faq.json
Comment (fynngodau) [reply src/data/faq.json:502]: @heinezen suggested to add this information twice in https://github.com/corona-warn-app/cwa-website/pull/1211#discussion_r627500614.

@MikeMcC399, I think you are right with considering the sentence without the additional sentence part starting with "as well as", and then it needs to be "Both […] are".

**Comment by Lu9Bro** [comment]
The revised text looks good to me, thanks.

In one section, the text reads "To resolve the problem, please ensure that you are using the latest version of your smartphone's operating system. If this doesn't help, please contact your device manufacturer."

Do you think it would help to add two sentences here like 
"(Reverting to an older version of CWA does not help, because the technical incompatibility existed in previous versions, too. Just it was not visible to the user.)"    ?

This is meant for users like in #1288 
> Seit dem letzten Update v. 10.05. ist mein Bluetooth Chip plötzlich inkompatibel. Das kann ja wohl nicht sein. Mit der Vorgängerversion wieder alles okay. Geht das bei der Corona App jetzt genauso wie bei den meisten Banking Apps? Jedes Update eine Verschlimmbesserung?

But I’m uncertain if this suggestion would be too much.

**Comment by fynngodau** [comment]
@Lu9Bro Since there is no official downgrade procedure in both Google Play as well as in the stock Android system, I think the information that the check was added in 2.2 is sufficient. However some users might try this anyway… what do the others think?

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:507]: "att" => "at"

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:513]: "assesment" => "assessment"

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:498]: "assesment" => "assessment"

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
I marked up some typos.

**Comment by fynngodau** [comment]
I have added a German translation now.

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json:496]: Maybe use the term "Startseite" displayed under the house icon by the app instead of "Hauptbildschirm"?

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json:509]: Same comment about "Hauptbildschirm" here too.

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Suggestion about using the term "Startseite".

File: src/data/faq_de.json
Comment (fynngodau) [reply src/data/faq_de.json:496]: Both terms are used throughout the FAQ, but I agree that Startseite, as a term, is a bit better.

File: src/data/faq.json
Comment (fynngodau) [inline src/data/faq.json:513]: Having thought about this again, since risk assessment is toggled together with Exposure Logging, you can't actually enable it (on devices without Bluetooth), can you?

Comment (MikeMcC399) [reply src/data/faq.json:513]: @fynngodau 
I can confirm on Pixel 3a API 29 emulator (which has no Bluetooth support) that it is **not** possible to activate exposure logging. Attempting to activate produces a Cause: 3 error. Clicking on the DETAILS button in the emulator produces no detailed error output.

This probably needs to be raised as an issue in https://github.com/corona-warn-app/cwa-app-android, because the text output "You can send and receive warnings that result from check-ins." by the app is inaccurate. I will leave that to you unless you say you want me to do something. Hopefully there will be very few users affected by this.

I don't have any physical Android device without Bluetooth to test on. The emulators cannot always be relied on to show the exact same behavior that a physical device shows, so please take the results of my emulator test with caution.

Comment (MikeMcC399) [reply src/data/faq.json:513]: I am seeing some entries in the log file such as
`2021-05-28T05:38:30.598Z  V/TracingRepository: TracingProgress: TracingProgress.IsCalculating, isExposureDetecting=false, isEWDownloading=false, isEWCalculatingRisk=true, isPTDownloading=false, isPTCalculatingRisk=false`
EW = Exposure Window
PT = Presence Tracing
I was not checked into any events in my test though.
Maybe it does work without activating exposure logging?

**Comment by heinezen** [comment]
@fynngodau I would take over now and make remaining in preparation for the next release. So if you still have local changes to push, the best time to do that is now :)

**Comment by fynngodau** [comment]
> So if you still have local changes to push, the best time to do that is now :)

This PR is all I have at the moment ;)

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:499]: "featue" => "feature"

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json:501]: "des Kontakttagebuch" => "des Kontakttagebuchs"

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
I marked up a couple of typos.

**Comment by heinezen** [comment]
I made quite a few changes to the existing text:

* More precise definition of the problem
* Merge the last paragraph (before the technical explanation) into the second paragraph
* Align text and wording between the two FAQ entries
* Remove first paragraph of `full_incompat` because it could be confusing
* Add technica explanation to `full_incompat`

File: src/data/faq.json
Comment (fynngodau) [inline src/data/faq.json:514]: This does not apply to 2 out of 3 possible problems that are mentioned above. This was the reason that I chose a different phrasing here.

But I think that in the third case, it is already clear to users what they are doing ;)

File: src/data/faq.json
Comment (fynngodau) [inline src/data/faq.json]: I understand that this is not a great introduction to the entry. However, I was worried that users could scroll down to the next entry by accident (also specifically since it now looks very similar), and falsely think that their device cannot bluetooth at all.

**Review by fynngodau — COMMENTED** [review]
I like your changes @heinezen. Here's some thoughts on two of them.

File: src/data/faq.json
Comment (MikeMcC399) [reply src/data/faq.json:514]: Perhaps instead of saying
"To resolve the problem"
write
"To resolve any software incompatibility" ?

Comment (heinezen) [reply src/data/faq.json:514]: I figured that the people who use Android Emulator could probably figure out that it's not a problem for them :D

File: src/data/faq.json
Comment (heinezen) [reply src/data/faq.json]: My problem with the introduction sentence is that for someone who reads this without knowing about the nature of the incompatibility, it would be even more confusing. Because then it is not really clear which variant (partial or full) actually applies to the user.

I agree that this could be a problem if a user scrolls down by accident or ends up reading the wrong entry when they use the FAQ search, but that's not something that can be easily fixed with more text and links. Making the warning messages distinct would be one way to solve it I guess.

**Comment by Ein-Tim** [comment]
I don't know what's the plan under Android but iOS version 2.3 has been released today.

**Pull request merged**
